Tasks/Common/TlsHelper_/TlsHelper_.psm1

```diff
@@ -0,0 +1,51 @@
+Import-VstsLocStrings -LiteralPath $PSScriptRoot/module.json
+
+function Add-Tls12InSession {
+    [CmdletBinding()]
+    param()
+
+    try {
+        if ([Net.ServicePointManager]::SecurityProtocol -notcontains 'Tls12') {
+            [Net.ServicePointManager]::SecurityProtocol += [Net.SecurityProtocolType]3072
+            Write-Verbose 'Added TLS 1.2 in session.'
+        }
+        else {
+            Write-Verbose 'TLS 1.2 already present in session.'
+        }
+    }
+    catch {
+        Write-VstsTaskError "Failed to add TLS 1.2 in session: $_.Exception.Message"
```

Comment:
You don't need to catch and throw. this would alter stack trace. You already added trace saying adding TLS1.2. any error just after that will automatically signify TLS error.

Reply:
Write-VstsTaskError does not throw. It only writes error in red. If setting TLS 1.2 in session fails, I don't want it to fail the task as it might not really be required.